server/src/main/java/org/opensearch/index/query/IntervalBuilder.java

```diff
@@ -146,7 +154,22 @@ protected static IntervalsSource combineSources(List<IntervalsSource> sources, i
         if (maxGaps == 0 && ordered) {
             return Intervals.phrase(sourcesArray);
         }
-        IntervalsSource inner = ordered ? Intervals.ordered(sourcesArray) : Intervals.unordered(sourcesArray);
+        IntervalsSource inner;
+        if (ordered) {
+            inner = Intervals.ordered(sourcesArray);
+        } else {
```

Comment:
collapse? 

```
if (ordered) {
} else if (overlap) {
} else {
}
```

Reply:
@dblock They are, we can make this an enum I believe.  Let me work on it.